## MLPP HW1 - Diagnostic - Part 3

### Naive 311 Predictions

In [1]:
import warnings
warnings.simplefilter('ignore')

%run helper.py
%matplotlib inline

In [2]:
# load datasets
buildings = pd.read_pickle('vacant_buildings_17.pkl')
graffiti = pd.read_pickle('graffiti_17.pkl')
lights = pd.read_pickle('alley_lights_17.pkl')

In [3]:
# drop non-meaningful values
buildings = drop_if(buildings, col_list=['Ward', 'ZIP CODE', 'Police District', 'Community Area'])
graffiti = drop_if(graffiti)
lights = drop_if(lights)
buildings = combine_address(buildings)

In [4]:
# Check for address
b_prob = search_by(buildings, 'Street Address', '3600 W Roosevelt Ave')
g_prob = search_by(graffiti, 'Street Address', '3600 W Roosevelt Ave')
l_prob = search_by(lights, 'Street Address', '3600 W Roosevelt Ave')

print('Prob of Vacant Building Request: {}'.format(b_prob))
print('Prob of Graffiti Removal Request: {}'.format(g_prob))
print('Prob of Out Alley Light Request: {}'.format(l_prob))

Prob of Vacant Building Request: 0.0
Prob of Graffiti Removal Request: 0.0
Prob of Out Alley Light Request: 0.0


I didn't find the address in the datasets, so I will take a step up and look at the zip code for this address, which is 60624. 

In [5]:
b_prob = search_by(buildings, 'ZIP CODE', '60624')
g_prob = search_by(graffiti, 'ZIP Code', '60624')
l_prob = search_by(lights, 'ZIP Code', '60624')

print('Prob of Vacant Building Request: {}'.format(b_prob))
print('Prob of Graffiti Removal Request: {}'.format(g_prob))
print('Prob of Out Alley Light Request: {}'.format(l_prob))

Prob of Vacant Building Request: 0.039836289222373804
Prob of Graffiti Removal Request: 0.0027585350493609245
Prob of Out Alley Light Request: 0.021945709470362534


Based on the zip code, it seems most likely that a request came in about a vacant or abandoned building for 3600 W Roosevelt. It seems least likely that the request was for graffiti removal.

In [6]:
# check for graffiti requests from uptown and garfield park
g_uptown = search_by(graffiti, 'ZIP Code', '60640|60613')
g_garfieldpark = search_by(graffiti, 'ZIP Code', '60612|60624')

print('Uptown prob: {}'.format(g_uptown))
print('Garfield Park prob: {}'.format(g_garfieldpark))
print('Difference: {}'.format(g_uptown - g_garfieldpark))

Uptown prob: 0.045413824606842235
Garfield Park prob: 0.020054815905482477
Difference: 0.025359008701359757


Based on the number of requests from these areas in 2017, it seems 2.5% more likely that a graffiti removal request came from uptown.

In [7]:
up = 400/1000
gp = 600/1000

up_g = 160/400
gp_g = 100/600

up_prob = (up*up_g) / ((gp*gp_g) + (up*up_g))
gp_prob = (gp*gp_g) / ((gp*gp_g) + (up*up_g))

print('Uptown prob: {}'.format(up_prob))
print('Garfield Park prob: {}'.format(gp_prob))
print('Difference: {}'.format(up_prob - gp_prob))

Uptown prob: 0.6153846153846155
Garfield Park prob: 0.3846153846153846
Difference: 0.23076923076923095


Based on these numbers, it seems 23% more likely that a graffiti removal request came from uptown. 